### PR TITLE
Player has 'stop' button while live

### LINKED ISSUE #287: use stop button instead of pause button on live stream
Currently you can pause the live stream, and play back. This is kind of unreliable as you can end up several seconds or minutes behind of the actual live stream.

Usually you just want to reload the stream and start it playing again anyway, so it would be better if it just 'reloaded' the stream again instead of pausing it, so you don't have to worry about reloading it manually to catch up to the current time.

So there are two parts to this:

* change the pause icon to a stop icon (playing back podcasts should still pause!)
* change the behavior to stop and reload the stream instead of pausing

### PR DESCRIPTION
**Pull request by sharpshark28**
Fixes #287

Let me know if this is the kind of fix you imagined. I'm not 100% sure how the player works or if this was working right as the live streamer just signed off after I started working on this hahaaaaaa. But in theory if the code comments are to be believed this may handle it.

image

### CODE REVIEW

**Comment by mcfiredrill** [comment]
Looks great! I will test it out this weekend and merge. Thank you so much again!! :tada: :rocket: :heart_decoration:

File: app/components/datafruits-player.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ export default Component.extend({
       audioTag.pause();
       this.set('playButtonPressed', false);
       this.set('playerState', 'paused');
-      if(this.playingPodcast === false){
+      if(this.playingPodcast === false || this.isLive){
```
Comment (mcfiredrill) [inline app/components/datafruits-player.js]: I'm not sure this is needed here. If we're playing *not* playing a podcast, it should always be a live stream that's playing instead.

File: app/templates/components/container/datafruits-player.hbs [hunk 2]
```diff
@@ -16,7 +16,7 @@
       </a>
     {{else if playing}}
       <a class="jp-pause" tabindex="1" {{action "pause"}}>
-        {{t "player.pause"}}
+        {{t (if this.isLive "player.stop" "player.pause")}}
```
Comment (mcfiredrill) [inline app/templates/components/container/datafruits-player.hbs]: So I think the `isLive` computed property is a bit of hack...it really only looks at the title to see if it starts with "LIVE". Its totally possible to change the title to something else, even if its still playing the live stream. 

I think in this case we can use the `playingPodcast` property to achieve this.

File: app/templates/components/datafruits-player.hbs [hunk 3]
```diff
@@ -21,7 +21,7 @@
       </a>
     {{else if playing}}
       <a class="jp-pause" tabindex="1" {{action "pause"}}>
-        {{t "player.pause"}}
+        {{t (if this.isLive "player.stop" "player.pause")}}
```
Comment (mcfiredrill) [inline app/templates/components/datafruits-player.hbs]: And the same for here.
```suggestion
        {{t (if this.playingPodcast "player.pause" "player.stop")}}
```

**Comment by sharpshark28** [comment]
Great feedback, I'll get to it when I can.

File: app/templates/components/datafruits-player.hbs (hunk 3 above)
Comment (sharpshark28) [reply app/templates/components/datafruits-player.hbs]: Addressed in 
7839b97

File: app/templates/components/container/datafruits-player.hbs (hunk 2 above)
Comment (sharpshark28) [reply app/templates/components/container/datafruits-player.hbs]: Addressed in 
7839b97

File: app/components/datafruits-player.js (hunk 1 above)
Comment (sharpshark28) [reply app/components/datafruits-player.js]: Addressed in 
7839b97

**Comment by sharpshark28** [comment]
@mcfiredrill also played with this while streaming today, I believe it works better to do the stream reload on Play rather than on Pause such that the current time is requested when the user is ready to begin again. Not 100% on this but it feels good while I'm listening today 5b84d8b

**Comment by mcfiredrill** [comment]
Really sorry for dropping the ball on this!
Only issue I'm seeing is the loading spinner never shows. For some reason the player state is always 4 , not 0 while its loading.
I will merge this and fix in another branch if its not a big deal to fix it myself.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState

**Pull request merged**